### ISSUE TITLE
Default values for Injected Link on Login Page have no effect

### ISSUE THREAD

**Issue by michael-milette** [IT-Kartellet/moodle-auth_unilogin#3]
Thank you for making this plugin available to the Moodle community.

I had a lot of trouble getting the injected link to appear on the login page until I realized that the default values for following two fields had no effect:
- the login_behaviour_link_text
- login_behaviour_link_selector

Once I copied the default values into the text input fields and saved, it started working correctly. I haven't played with the rest of the fields yet to see if they have similar issues but not having default values have any effect was quite confusing. 

Could you modify your code to ensure that default values have an effect if no value was specified?

Best regards,

Michael

**Comment by janmeier** [comment]
Yes, I actually believed that using the `admin_setting_configtextarea` etc. classes would automatically insert the default values into the fields, but appearently I was wrong. 

I also remember that some modules present you with a "these settings are new, please do something about them" page when upgrading a module so I was hoping that moodle would show the settings when installing, but that's appearently not the case... 

I'll get on it

**Comment by michael-milette** [comment]
Try putting this in your loginpage_hook() function:

```
    $unilogin = optional_param('unilogin', 'true', PARAM_ALPHA);
    if (empty($CFG->alternateloginurl) && $this->config->login_behaviour === 'redirect' && $unilogin != 'false') {
        // login/index.php will automatically redirect based on this alternate login url.
        $CFG->alternateloginurl = $this->get_url();
    } else {
        // Get link text.
        $link_text = format_string($this->config->login_behaviour_link_text);
        if (empty($link_text)) {
            $link_text = get_string('login_behaviour_link_text_default', 'auth_unilogin');
        }
        // Get link selector.
        $link_selector = format_string($this->config->login_behaviour_link_selector);
        if (empty($link_selector)) {
            $link_selector = '#login';
        }
        $PAGE->requires->yui_module(
            'moodle-auth_unilogin-link_injector',
            'M.auth_unilogin.link_injector.init',
            array($this->get_url(), $link_text, $link_selector)
        );
    }
```

**Comment by janmeier** [comment]
In know how to fix it, I'm just saying, that I thought moodle had some kind of handling for this already :) 

But don't worry, I'll get on it

**Comment by michael-milette** [comment]
I had no doubt that you could fix it. Just trying to help out.

**Comment by janmeier** [comment]
Thanks for the feeback, this has been published as v1.0.2 https://moodle.org/plugins/pluginversion.php?id=8184

**Comment by michael-milette** [comment]
Thanks for the quick fix Jan!
